cloud-stream-dynamic-destination-processor/src/main/java/com/solace/samples/spring/scs/DynamicDestinationProcessor.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.solace.samples.spring.scs;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.stream.function.StreamBridge;
+import org.springframework.context.annotation.Bean;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+
+@SpringBootApplication
+public class DynamicDestinationProcessor {
+
+	private static final Logger log = LoggerFactory.getLogger(DynamicDestinationProcessor.class);
+	private static AtomicInteger counter = new AtomicInteger(0);
+
+	public static void main(String[] args) {
+		SpringApplication.run(DynamicDestinationProcessor.class);
+	}
+
+	/*
+	 * Dynamic Topic Publish OPTION 1: Using scst_targetDestination header; works
+	 * with any binder that supports the header, including Solace.
+	 * 
+	 * Note that the `scst_targetDestination` header is essentially telling the
+	 * binder to override the default destination specified on a binding and if the
+	 * header is NOT set then the message would be sent to the default destination.
+	 */
+	@Bean
+	public Function<Message<String>, Message<String>> functionUsingTargetDestHeader() {
+		return input -> {
+			String topic = getMyTopicUsingLogic(input.getPayload());
+			log.info("Processing message: " + input.getPayload());
+			String payload = input.getPayload().concat(" Processed by functionUsingTargetDestHeader");
+			return MessageBuilder.withPayload(payload).setHeader("scst_targetDestination", topic).build();
```

Comment:
Instead of explicitly hard-coding `scst_targetDestination`, it would be better to use `BinderHeaders.TARGET_DESTINATION`.

Reply:
Great point! Updated :)